scripts/utils.py

```diff
@@ -337,3 +344,403 @@ def add_population_to_dataframe(
+      min_frac_individual_population with respect to the total population of the region.
+    2. Among those, sort countries by descending population, and cut as soon as the cumulative population exceeds
+      min_frac_cumulative_population.
+    Note: It may not be possible to fulfil both conditions. In that case, a warning is raised.
+
+    Parameters
+    ----------
+    region : str
+        Name of the region.
+    reference_year : int
+        Reference year to consider when selecting countries.
+    min_frac_individual_population : float
+        Minimum fraction of the total population of the region that each of the listed countries must exceed.
+    min_frac_cumulative_population : float
+        Minimum fraction of the total population of the region that the sum of the listed countries must exceed.
+    countries_regions : pd.DataFrame or None
+        Countries-regions dataset, or None, to load it from owid catalog.
+    population : pd.DataFrame or None
+        Population dataset, or None, to load it from owid catalog.
+
+    Returns
+    -------
+    countries : list
+        List of countries that are expected to have the largest contribution.
+
+    """
+    # TODO: Add tests.
+    if countries_regions is None:
+        countries_regions = catalog.find(
+            "countries_regions", dataset="reference", namespace="owid"
+        ).load()
+
+    if population is None:
+        population = (
+            catalog.find("population", namespace="owid", dataset="key_indicators")
+            .load()
+            .reset_index()
+        )
+
+    # List all countries in the selected region.
+    members = list_countries_in_region(region, countries_regions=countries_regions)
+
+    # Select population data for reference year for all countries in the region.
+    reference = (
+        population[
+            (population["country"].isin(members))
+            & (population["year"] == reference_year)
+        ]
+        .dropna(subset="population")
+        .sort_values("population", ascending=False)
+        .reset_index(drop=True)
+    )
+
+    # Calculate total population in the region, and the fractional contribution of each country.
+    total_population = reference["population"].sum()
+    reference["fraction"] = reference["population"] / total_population
+
+    # Select countries that exceed a minimum individual fraction of the total population of the region.
+    selected = reference[
+        (reference["fraction"] > min_frac_individual_population)
+    ].reset_index(drop=True)
+
+    # Among remaining countries, select countries that, combined, exceed a minimum fraction of the total population.
+    selected["cumulative_fraction"] = selected["population"].cumsum() / total_population
+    candidates_to_ignore = selected[
+        selected["cumulative_fraction"] > min_frac_cumulative_population
+    ]
+    if len(candidates_to_ignore) == 0:
+        print(
```

Comment:
No need to do it now, but we might use proper logging when moving it to data-utils. E.g. with structlog that colors it nicely in the terminal
```
log.warning("Conditions are too strict to select countries that must be included in the data.")
```
or
```
log.info("countries must be informed with enough % if the population; otherwise aggregate data will be nan", region=region, coverage=selected['fraction'].sum(), n_countries=len(selected))
```

Reply:
Yes, let's do that when moving to data-utils.